src/Controller/Articles/ShowController.php

```diff
@@ -25,100 +21,50 @@ public function __invoke(
-        }
-
-        $articles = $isiteResult->getDomainModels();
-        if (!$articles) {
-            throw $this->createNotFoundException('No articles found for guid');
-        }
-
-        /** @var Article $article */
-        $article = reset($articles);
-
-        if ($slug !== $article->getSlug()) {
-            return $this->redirectWith($article->getKey(), $article->getSlug(), $preview);
-        }
-        if ($article->getBbcSite()) {
-            $this->setIstatsExtraLabels(['bbc_site' => $article->getBbcSite()]);
-        }
-        $context = null;
-        $parentProgramme = null;
-        $projectSpace = $article->getProjectSpace();
-        if (!empty($article->getParentPid())) {
-            $context = $coreEntitiesService->findByPidFull($article->getParentPid());
-            if ($context instanceof Group) {
-                $parentProgramme = $context->getParent();
+        } catch (HasContactFormException $e) {
+            if (!$this->slug) {
+                $route = 'article_with_contact_form_noslug';
+                $params = ['key' => $this->key];
             } else {
-                $parentProgramme = $context;
-            }
-            if ($context && ($article->getProjectSpace() !== $context->getOption('project_space'))) {
-                throw $this->createNotFoundException('Project space Article-Programme not matching');
+                $route = 'article_with_contact_form';
+                $params = ['key' => $this->key, 'slug' => $this->slug];
             }
-        }
-        $this->overrideMetaTagsValues($article);
-        $this->setContext($context);
-        $this->setAtiContentId($guid, 'isite2');
 
-        if ('' !== $article->getBrandingId()) {
-            $this->setBrandingId($article->getBrandingId());
+            return $this->cachedRedirectToRoute($route, $params, 302, 3600);
         }
 
-        $parents = $article->getParents();
-        $siblingPromise = $isiteService->setChildrenOn($parents, $article->getProjectSpace()); //if more than 48, extras are removed
-        $childPromise = $isiteService->setChildrenOn([$article], $article->getProjectSpace(), $this->getPage());
-        $response = $this->resolvePromises(['children' => $childPromise, 'siblings' => $siblingPromise]);
+        $this->initPersonalisation();
 
+        $parents = $this->isiteObject->getParents();
+        $siblingPromise = $isiteService->setChildrenOn($parents, $this->isiteObject->getProjectSpace()); //if more than 48, extras are removed
+        $childPromise = $isiteService->setChildrenOn([$this->isiteObject], $this->isiteObject->getProjectSpace(), $this->getPage());
+        $response = $this->resolvePromises(['children' => $childPromise, 'siblings' => $siblingPromise]);
+        $children = reset($response['children']);
+        $paginatorPresenter = null;
+        if ($children) {
+            $paginatorPresenter = $this->getPaginator(reset($response['children'])->getTotal());
```

Comment:
```suggestion
            $paginatorPresenter = $this->getPaginator($children->getTotal());
```
calling reset(...) seems pointless